dallinger/experiment_server/experiment_server.py

```diff
@@ -319,7 +319,10 @@ def advertisement():
         status = part.status
     except exc.SQLAlchemyError:
         status = None
-
+    try:
+        app_id = config.get('id')
```

Comment:
I don't understand the distinction you're making. This code is setting app_id to 'unknown' if a value isn't specified, which seems like a default value to me.